lib/maths/CBayesianOptimisation.cc

```diff
@@ -372,6 +398,268 @@ double CBayesianOptimisation::kernel(const TVector& a, const TVector& x, const T
+                LOG_ERROR(<< "Unexpected name for restoring bayesian optimization parameters: "
+                          << traverser.name());
+                return false;
+            }
+        } while (traverser.next());
+    } catch (std::exception& e) {
+        LOG_ERROR(<< "Failed to restore state! " << e.what());
+        return false;
+    }
+
+    return true;
+}
+
+bool CBayesianOptimisation::restoreSubLevelVector(const std::string& tag,
+                                                  const std::string& name,
+                                                  TVector& vector,
+                                                  core::CStateRestoreTraverser& traverser) {
+    if (traverser.hasSubLevel()) {
+        if (traverser.traverseSubLevel(restoreVector(tag, vector)) == false) {
+            return false;
+        }
+    } else {
+        LOG_ERROR(<< "Error restoring " << name << ": " << traverser.value()
+                  << ". Expected to have a sublevel");
+        return false;
+    }
+    return true;
+}
+
+bool CBayesianOptimisation::restoreSubLevelVector(const std::string& tag,
+                                                  const std::string& name,
+                                                  TDoubleVec& vector,
+                                                  core::CStateRestoreTraverser& traverser) {
+    if (traverser.hasSubLevel()) {
+        if (traverser.traverseSubLevel(restoreVector(tag, vector)) == false) {
+            return false;
+        }
+    } else {
+        LOG_ERROR(<< "Error restoring " << name << ": " << traverser.value()
+                  << ". Expected to have a sublevel");
+        return false;
+    }
+    return true;
+}
+
+void CBayesianOptimisation::acceptPersistInserter(core::CStatePersistInserter& inserter) const {
+    try {
+        inserter.insertLevel(MIN_BOUNDARY_TAG, persistVector(BOUNDARY_TAG, m_MinBoundary));
+        inserter.insertLevel(MAX_BOUNDARY_TAG, persistVector(BOUNDARY_TAG, m_MaxBoundary));
+        inserter.insertLevel(ERROR_VARIANCES_TAG,
+                             persistVector(ERROR_VARIANCE_TAG, m_ErrorVariances));
+        inserter.insertLevel(KERNEL_PARAMETERS_TAG,
+                             persistVector(KERNEL_PARAMETER_TAG, m_KernelParameters));
+        inserter.insertLevel(MIN_KERNEL_COORDINATE_DISTANCE_SCALES_TAG,
+                             persistVector(MIN_KERNEL_COORDINATE_DISTANCE_SCALE_TAG,
+                                           m_MinimumKernelCoordinateDistanceScale));
+        inserter.insertLevel(FUNCTION_MEAN_VALUES_TAG,
+                             persistFunctionMeanValues(m_FunctionMeanValues));
+    } catch (std::exception& e) {
+        LOG_ERROR(<< "Failed to persist state! " << e.what());
+    }
+}
+
+std::function<void(core::CStatePersistInserter&)>
+CBayesianOptimisation::persistVector(const std::string& tag,
+                                     const CBayesianOptimisation::TVector& vector) const {
```

Comment:
As above have a look at uses of `CPersistUtils`.

Reply:
Changed in a1f55433acfe120c4b58ba7836f64fc6b2cf535c